files/json_output_formatter.rb

```diff
@@ -15,19 +15,24 @@ def dump_summary(summary)
       map { |example| example.metadata[:points].to_i }.
       sum
 
+    score = (earned_points.to_f / total_points).round(4)
+    if score.nan?
+      score = 0
+    end
+    
     @output_hash[:summary] = {
       duration: summary.duration,
       example_count: summary.example_count,
       failure_count: summary.failure_count,
       pending_count: summary.pending_count,
       total_points: total_points,
       earned_points: earned_points,
-      score: (earned_points.to_f / total_points).round(4)
+      score: score
     }
     score = (@output_hash[:summary][:score] * 100).round(2)
     
-    if score.nan?
-      score = "This project is not graded."
+    if summary.example_count.zero?
+      score = "An error occurred while running tests"
     else
       score = score.to_s + "%"
```

Comment:
Oy this whole thing is feeling funny to me. All of this transformation and formatting being done by grade_runner, and then sending back a string... grade_runner should simply send the output of the test runner to grades, and grades should handle it from there. But let's solve our immediate problem for now.

Reply:
@raghubetina  It does feel a little silly! I think probably what should happen is that the formatter should capture the failures outside of examples count like the regular json formatter does.

```bash
{
  "version":"3.9.2",
  "messages":["No examples found."],
  "examples":[],
  "summary": {
    "duration":4.7e-05,
    "example_count":0,
    "failure_count":0,
    "pending_count":0,
    "errors_outside_of_examples_count":1
  },
  "summary_line":"0 examples, 
  0 failures, 
  1 error occurred outside of examples"
}% 
```

Reply:
Yes this seems right. However, for now let's just solve our immediate problem; we can discuss whether we want to tackle a structural change or wait until 2.0.